datafusion/expr/src/expr.rs

```diff
@@ -957,22 +971,18 @@ fn create_name(e: &Expr) -> Result<String> {
             );
             Ok(s)
         }
-        Expr::Case {
-            expr,
-            when_then_expr,
-            else_expr,
-        } => {
+        Expr::Case(case) => {
```

Comment:
If you wanted to avoid `&case` I think you could do like

```suggestion
        Expr::Case(Case {
            expr,
            when_then_expr,
            else_expr,
        }) => {
```

Maybe the `&` is nicer syntax 🤔

Reply:
I debated between the two styles too. I don't have a strong preference either way